### PR TITLE
[CLB-22] Show/Hide AddNewSkill modal done

### PR DESCRIPTION
**Pull request by rocky1207**

### CODE REVIEW

File: src/redux/reducers/modalReducer.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Action } from 'redux';
+import actionTypes from '../actions/actionTypes';
+
+const initialState = {
+    show: false,
+};
+
+const modalReducer = (state = initialState, action: Action) => {
```
Comment (kimnovak) [inline src/redux/reducers/modalReducer.tsx]: We will have different types of Modals in the app, so a modal type should be added to the state

File: src/redux/actions/actionTypes.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 const actionTypes = {
     ADD: 'ADD',
     SUB: 'SUB',
+    OPEN: 'OPEN',
+    CLOSE: 'CLOSE',
 };
```
Comment (kimnovak) [inline src/redux/actions/actionTypes.ts]: OPEN and CLOSE are a bit too generic, can you rename it to OPEN_MODAL and CLOSE_MODAL?

File: src/redux/actions/counter.ts [hunk 3]
```diff
@@ -11,3 +11,14 @@ export const sub = () => {
...
+export const open = () => {
+    return {
+        type: actionTypes.OPEN,
+    };
+};
+export const close = () => {
+    return {
+        type: actionTypes.CLOSE,
+    };
+};
```
Comment (kimnovak) [inline src/redux/actions/counter.ts]: You should create a modal.ts file inside this folder and move these 2 actions there

File: src/redux/reducers/modalReducer.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    switch (action.type) {
+        case actionTypes.OPEN:
+            state.show = true;
+            return Object.assign({}, state);
```
Comment (kimnovak) [inline src/redux/reducers/modalReducer.tsx]: Shorter / easier syntax that doesn't mutate the state
```suggestion
            return {
                ...state,
                show: true
            }
```

File: src/redux/reducers/modalReducer.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+        case actionTypes.CLOSE:
+            state.show = false;
+            return Object.assign({}, state);
```
Comment (kimnovak) [inline src/redux/reducers/modalReducer.tsx]:
```suggestion
            return {
                 ...state,
                 show: false
             }
```

**Review by kimnovak — CHANGES_REQUESTED** [review]

**Comment by kimnovak** [comment]
Can you also copy the package-lock.json we have here: https://raw.githubusercontent.com/q-praksa/collaborator/development/package-lock.json and paste it inside your local package-lock.json, and add to commit to avoid the conflicts

**Review by vukstojanovic — APPROVED** [review]

**Review by SreckoStojic — APPROVED** [review]
Great job!

**Pull request merged**